### PR TITLE
feat: [CGC-6] Ellipsis button for the chart

### PR DESCRIPTION
**Pull request by petterav**

### CODE REVIEW

File: src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        modifiers: [
+            {
+                name: 'offset',
+                options: { offset: [0, 5] },
+            },
+        ],
+    });
+
+    return (
+        <div className='absolute top-2 right-2'>
```
Comment (petterav) [inline src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx:27]: this is just to show how the styling will be like, ofc we wont have it "into" the graph, but i mean that this should be a part of the component

File: src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            <EllipsisButton onClick={() => setIsVisible(!isVisible)} isVisible={isVisible} setReferenceElement={setReferenceElement} />
```
Comment (edvinstava) [inline src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx:28]: Use previous state to ensure that the state updates on the most recent state value, minimizing the risk of "hook desync".
```suggestion
            <EllipsisButton onClick={() => setIsVisible(prevState => !prevState)} isVisible={isVisible} setReferenceElement={setReferenceElement} />
```

File: src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            {isVisible && (
+                <PopoverList
+                    setPopperElement={setPopperElement}
+                    style={styles.popper}
+                    popoverAttributes={{ ...attributes.popper }}
+                >
+                    <PopoverListItem label="Settings" icon={<span>🔧</span>} onClick={() => console.log('Not Implemented')} />
+                    <PopoverListItem label="Annotations" icon={<span>📥</span>} onClick={() => setShowAnnotation(prevState => !prevState)} />
```
Comment (edvinstava) [inline src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx:36]: prevState here is undefined..

File: src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        placement: 'bottom',
+        modifiers: [
+            {
+                name: 'offset',
+                options: { offset: [0, 5] },
+            },
+        ],
+    });
+
+    return (
```
Comment (edvinstava) [inline src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx:26]: Can you make the Popover list close if the page is clicked somewhere else?

File: src/Components/GrowthChart/ChartSettingsButton/PopoverList/PopoverListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+type PopoverListItemProps = {
+  label: string;
+  icon: React.ReactNode;
+  onClick: () => void;
+};
+
+export const PopoverListItem = ({ label, icon, onClick }: PopoverListItemProps): JSX.Element => (
+  <div 
+  className="flex flex-row cursor-pointer hover:bg-gray-100 py-1 px-2 rounded min-w-fit"
```
Comment (edvinstava) [inline src/Components/GrowthChart/ChartSettingsButton/PopoverList/PopoverListItem.tsx:11]: dosnt look like this hover:bg-gray-100 is working, would be nice to have the background change on hover

**Comment by henrikmv** [comment]
@petterav, would you please fix the master conflicts before my review?

**Comment by henrikmv** [comment]
Make sure the changes actually show up in your branch. I can't see any ellipsis button. There is a lot of commented code, some of which should not be part of a branch set ready for review. Can you take a look at it @petterav ? Thanks

File: src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                    <PopoverListItem
+                        label='Print'
+                        icon={<span>🖨️</span>}
+                        onClick={() => {}}
+                    />
+                    <PopoverListItem
+                        label='Help'
+                        icon={<span>❓</span>}
+                        onClick={() => {}}
+                    />
```
Comment (edvinstava) [inline src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx:77]: You can consider removing some of these PopoverListItems as none of them are functional. Instead just have one that says "Coming soon" or something simular.
```suggestion
                    <PopoverListItem
                        label='Coming soon'
                        icon={<span>🔧</span>}
                        onClick={() => {}}
                    />
```

**Review by edvinstava — COMMENTED** [review]
Well done Petter, take a look at the comments

**Review by edvinstava — APPROVED** [review]
Well done Petter, LGTM!

**Pull request merged**